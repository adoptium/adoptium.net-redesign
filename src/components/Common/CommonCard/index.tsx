import React from "react"
import { Link } from "../../Link"

const CommonCard = ({ data }) => {
  return (
    <div className="p-10 flex flex-col max-w-[400px] newscard-2 h-[440px] md:h-[480px] bg-[#200E46]">
      <div className="grow">
        <h3 className="text-white text-[30px] font-semibold  whitespace-nowrap  md:whitespace-normal leading-[120%] md:text-[40px]">
          {data.title}
        </h3>
        <p className="mt-6 text-xl text-grey font-normal">{data.description}</p>
      </div>
      <Link to={data.href}>
        <button className="bg-transparent   border-2 border-pink-500/0 text-white text-base leading-6 font-normal w-[180px] h-[48px] rounded-[80px] gradient-border ">
          {data.button}
        </button>
      </Link>
    </div>
  )
}

export default CommonCard
