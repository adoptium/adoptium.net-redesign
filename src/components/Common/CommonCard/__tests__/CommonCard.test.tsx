import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import CommonCard from "../index"

describe("CommonButton component", () => {
  it("should render correctly", () => {
    const { container } = render(
        <CommonCard
            data={{
                title: "test",
                description: "test",
                button: "test",
            }}
        />
    )

    expect(container).toMatchSnapshot()
  })
})
