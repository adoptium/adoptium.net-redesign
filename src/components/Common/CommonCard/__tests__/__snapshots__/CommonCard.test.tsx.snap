// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommonButton component > should render correctly 1`] = `
<div>
  <div
    class="p-10 flex flex-col max-w-[400px] newscard-2 h-[440px] md:h-[480px] bg-[#200E46]"
  >
    <div
      class="grow"
    >
      <h3
        class="text-white text-[30px] font-semibold  whitespace-nowrap  md:whitespace-normal leading-[120%] md:text-[40px]"
      >
        test
      </h3>
      <p
        class="mt-6 text-xl text-grey font-normal"
      >
        test
      </p>
    </div>
    <a
      href="/enundefined"
    >
      <button
        class="bg-transparent   border-2 border-pink-500/0 text-white text-base leading-6 font-normal w-[180px] h-[48px] rounded-[80px] gradient-border "
      >
        test
      </button>
    </a>
  </div>
</div>
`;
